src/components/common/coordsValidate.ts

```diff
@@ -1,5 +1,46 @@
 export function parseCoords(coord): string[] {
   const regExp = /^(\-?\d+(?:\.\d+)?),\s*(\-?\d+(?:\.\d+)?)$/;
   const result = regExp.exec(coord);
-  return result !== null ? result.slice(1) : null;
+  if (result !== null) {
+    return result.slice(1);
+  } else {
+    //it's geopoint/geostring type
+    let geoObject;
+    try {
+      geoObject = JSON.parse(coord);
+    } catch (e) {
+      throw new Error("Invalid GeoJSON object.");
+    }
+    const hasType = geoObject.hasOwnProperty("type");
+    if (hasType) {
+      switch (geoObject.type) {
+        case "Point":
+        case "MultiPoint":
+        case "LineString":
+        case "MultiLineString":
+        case "Polygon":
+        case "MultiPolygon":
+          return geoObject.coordinates;
+        case "FeatureCollection":
+          return geoObject.features;
+        /*
+        default:
+          throw new Error('Invalid GeoJSON object.');
+          */
+      }
+    } else {
+      const hasGeography = geoObject.hasOwnProperty("Geography");
+      if (hasGeography) {
+        return geoObject;
+      } else {
+        const isSDTArray =
+          geoObject[0].hasOwnProperty("Geography") && Array.isArray(geoObject);
+        if (isSDTArray) {
+          geoObject["IsLayersArray"] = true;
+          return geoObject;
+        }
+      }
+    }
+    return geoObject;
```

Comment:
Maybe we could extract this logic to a method focused on parsing a GeoJson.
What do you think? That way, the comment would not be needed, and this block of code would be self-described by the method name.

Reply:
Did would be a nice improvement and would make the code more readable. What do you think @fedes1987 ?

Reply:
I agree. 
Addressed in 33b256a